docs/toolchain.md

```diff
@@ -0,0 +1,65 @@
+
+
+## Elección runtime
+
+A la hora de establecer la toolchain, se comienza por determinar cual será el entorno de ejecución de nuestra aplicación. Entre los entornos más conocidos para TypeScript encontramos:
+
+-Node.js: Aunque fue diseñado para JavaScript, admite TypeScript de forma nativa, es muy utilizado en aplicaciones del lado del servidor y es ampliamente conocido.
+
+-Deno: Parte de la misma base que Node.js, pero tiene algunas características que lo diferencian como los módulos de seguridad, la gestión de dependencias mediantes URLs, y la compatibilidad con TypeScript de forma nativa.
+
+Siguiendo los criterios de selección establecidos:
+
+Tanto Deno como Node presentan aproximadamente el mismo rendimiento, ya que comparten el mismo motor de JavaScript (V8). Si bien Deno es más seguro que Node, este último presenta una mayor y mejor documentación, lo que reduce la probabilidad de complicaciones en el proyecto. Además como veremos má adelante favorecerá la elección del gestor de dependencias, ya que se desarrolla en conjunto con NPM, maximizando la integración de estas dos herramientas.
+
+La elección final será Node.
+
+
+## Elección gestor de dependencias
+
+Una vez escogido el runtime de nuestro proyecto, debemos elegir el gestor de dependencias. Entre los pricipales y más usados tenemos:
+
+-NPM (Node Package Manager): Es el gestor de dependencias por defecto de Node.js. Es el más extendido entre la comunidad y tiene muchísimos paquetes públicos como resultado.
+
+-Yarn: Nace como respuesta por parte de Facebook a npm, por lo que sigue siendo compatible con Node.js. 
+
+-PNPM: Es un gestor minimalista centrado en la eficiencia.
+
+
+Siguiendo los criterios de selección establecidos:
+
+Si bien PNPM es un gestor interesante, se encuentra muy centrado en la optimización de los recursos usados, algo que no buscamos en nuestro proyecto, es el que tiene menor integración y menor documentación. Yarn es más seguro que NPM y más rápido y eficiente manejando paquetes, aunque no es compatible con los paquetes de NPM. NPM tiene mayor integración con Node (por definición) y, aunque posee un rendimiento ligeramente menor, se podrían considerar igualados en este aspecto teniendo en cuenta el proyecto en el que trabajamos. Lógicamente, existe más información disponible y más facilidades para su uso y configuración para NPM que para Yarn, al ser el primero más antiguo y desarrollado en conjunto con Node.
+
+La elección final será NPM.
+
+## Elección task runner
+
+Elegidos el runtime y gestor de dependencias, queda escoger el task runner. De entre los task runners más accesibles y con buena integración con nuestras herramientas tenemos:
+
+-NPM Scripts: El gestor de paquetes NPM tiene la posibilidad de ejecutar scripts personalizados.
+
+-Gulp: Es uno de los más populares, permite automatizar casi cualquier tarea de construcción. Está diseñado para crear y automatizar tareas mediante JavaScript.
```

Comment:
Creo que llegado a este punto conviene que conozcáis [snyk advisor](https://snyk.io/advisor/npm-package/gulp), que tiene una "puntuación de salud" para cada uno de los paquetes en npmjs. ¿Qué te parece que dice de Gulp? ¿Y de grunt?

Reply:
Lo he usado en esta nueva revisión, aunque tiene en cuenta métricas que en un principio nos serán irrelevantes (popularidad, comunidad), si he tenido en cuenta el mantenimiento, que puede generar deudas tecnicas a largo plazo.

Reply:
Si el valor es por debajo de 80, puede ser razón suficiente para ni siquiera considerarlo.
En todo caso, si se ha resuelto tienes que indicar el commit en el que se ha hecho y resolver.

Reply:
Hecho, he referenciado el commit que resuelve cada issue.

Reply:
¿Dónde? Tendrías que haberlo hecho aquí...

Reply:
Había puesto el comentario en el propio issue. Resuelto en #https://github.com/MarioGuisado/TrainMe/pull/34/commits/b4a4c3ee04b34d8c71d357deda5677792de1c548

Reply:
Lo correcto habría sido eliminarlo. No es cierto que sea uno de los más populares, de hecho. Puede que lo fuera hace 5 años.